cdap-api/src/main/java/co/cask/cdap/api/dataset/table/RowColumns.java

```diff
@@ -43,16 +48,19 @@
 
   public RowColumns(byte[] row) {
     this.row = row;
-    this.columns = new ArrayList<>();
   }
 
   public RowColumns(byte[] row, byte[]... columns) {
     this(row);
+    this.columns = new ArrayList<>();
     add(columns);
   }
 
   @SuppressWarnings("unchecked")
   public T add(byte[]... columns) {
+    if (this.columns == null) {
```

Comment:
A method that takes in `byte[][]` as Terence suggested would handle that, right?